catalog/validator/catalog-validator-metacardlocation/src/main/java/org/codice/ddf/validator/metacard/location/MetacardLocationValidator.java

```diff
@@ -0,0 +1,113 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.validator.metacard.location;
+
+import java.io.Serializable;
+import java.text.ParseException;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang.StringUtils;
+import org.locationtech.spatial4j.context.SpatialContext;
+import org.locationtech.spatial4j.context.jts.JtsSpatialContextFactory;
+import org.locationtech.spatial4j.exception.InvalidShapeException;
+import org.locationtech.spatial4j.io.WKTReader;
+
+import com.google.common.collect.ImmutableSet;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.data.types.Validation;
+import ddf.catalog.validation.MetacardValidator;
+import ddf.catalog.validation.ReportingMetacardValidator;
+import ddf.catalog.validation.ValidationException;
+import ddf.catalog.validation.impl.ValidationExceptionImpl;
+import ddf.catalog.validation.impl.report.MetacardValidationReportImpl;
+import ddf.catalog.validation.impl.violation.ValidationViolationImpl;
+import ddf.catalog.validation.report.MetacardValidationReport;
+import ddf.catalog.validation.violation.ValidationViolation;
+
+public class MetacardLocationValidator implements MetacardValidator, ReportingMetacardValidator {
+    private static final JtsSpatialContextFactory JTS_SPATIAL_CONTEXT_FACTORY =
+            new JtsSpatialContextFactory();
+
+    private static final String ERROR_MSG = Core.LOCATION + " is invalid: ";
+
+    private static final Set<String> VALIDATED_ATTRIBUTES = ImmutableSet.of(Core.LOCATION);
+
+    static {
+        JTS_SPATIAL_CONTEXT_FACTORY.allowMultiOverlap = true;
+    }
+
+    private static final SpatialContext SPATIAL_CONTEXT =
+            JTS_SPATIAL_CONTEXT_FACTORY.newSpatialContext();
+
+    @Override
+    public Optional<MetacardValidationReport> validateMetacard(Metacard metacard) {
+        MetacardValidationReportImpl report = null;
+
+        if (StringUtils.isNotEmpty(metacard.getLocation())) {
+            WKTReader wktReader = new WKTReader(SPATIAL_CONTEXT, JTS_SPATIAL_CONTEXT_FACTORY);
+            try {
+                wktReader.parse(metacard.getLocation());
+            } catch (ParseException | InvalidShapeException e) {
+                String message = ERROR_MSG + metacard.getLocation();
+                ValidationViolation violation = new ValidationViolationImpl(VALIDATED_ATTRIBUTES,
+                        message,
+                        ValidationViolation.Severity.ERROR);
+                report = new MetacardValidationReportImpl();
+                report.addMetacardViolation(violation);
+            }
+        } else {
+            Attribute validationErrorAttr = metacard.getAttribute(Validation.VALIDATION_ERRORS);
```

Comment:
Not sure I understand the need for this else block. Seems like we are trying to invalidate the metacard if it has been invalid before but since it has no location how is it still invalid? Probably just need some explanation of the bigger picture here.